miio/tests/test_vacuums.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022
+# Author: 2pirko
+import collections
+import sys
+from typing import List, Sequence, Tuple, Type
+
+import pytest
+
+from miio.device import Device
+from miio.integrations.vacuum.roborock.vacuum import ROCKROBO_V1
+from miio.interfaces import VacuumInterface
+
+from .test_device import DEVICE_CLASSES
+
+# list of all supported vacuum classes
+VACUUM_CLASSES: Tuple[Type[VacuumInterface], ...] = tuple(
+    cl for cl in DEVICE_CLASSES if issubclass(cl, VacuumInterface)  # type: ignore
+)
+
+
+def _all_vacuum_models() -> Sequence[Tuple[Type[Device], str]]:
+    """:return: list of tuples with supported vacuum models with corresponding class"""
+    result: List[Tuple[Type[Device], str]] = []
+    for cls in VACUUM_CLASSES:
+        assert issubclass(cls, Device)
+        vacuum_models = cls.supported_models
+        if sys.version_info >= (3, 10):
+            assert isinstance(vacuum_models, collections.abc.Iterable)
+        else:
+            assert isinstance(vacuum_models, collections.Iterable)
```

Comment:
Is this a real issue among the currently supported versions? If yes, it would be better to perform this check on import time:
```
try:
    from collections.abc import Iterable
except <exception type>:  # for python <3.9
    from collections import Iterable
```